File: Problem sets/pset6 (Some problem sets but in python)/readability/readability.py
from cs50 import get_string

# prompt user for text, remove white spaces, declare variables for number or words, sentences and letters and initialize to 0
s = get_string("text: ").strip()

number_words = 0
number_sentences = 0
number_letters = 0

for i in range(len(s)):
    # determine letters by checking if its an alphabet and increase letter count by 1 for every letter detected

    if s[i].isalpha():
        number_letters += 1

# determine words by checking for blank spaces. characters after blank spaces indicate the begining of a new word, hence word count will be increased by 1

    if (i == 0 and s[i] != ' ') or (i != len(s) - 1 and s[i] == ' ' and s[i + 1] != ' '):
        number_words += 1

# determine what are sentences by using ". ! and ?" as indicators of end of sentences. When a sentence is detected increment the sentence count by 1

    if s[i] == '.' or s[i] == '?' or s[i] == '!':
        number_sentences += 1

# solve for L, S and index respectively and round the result of index to a whole number

L = number_letters / number_words * 100
S = number_sentences / number_words * 100

index = round(0.0588 * L - 0.296 * S - 15.8)

# if the result of index is less than 1 print Before grade 1, if greater than 16 print Grade 16+, for anything else, the value of index should be printed alongside "Grade"

if index < 1:
    print("Before Grade 1")

elif index >= 16:
    print("Grade 16+")

else:
    print(f"Grade {index}")

File: Problem sets/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


int main(void)
{
// prompt user for text, declare variables for number or words, sentences and letters and initialize to 0
    string s = get_string("text: ");
    int number_words;
    int number_sentences;
    int number_letters;

    number_words = 0;

    number_sentences = 0;

    number_letters = 0;


    for (int i = 0; i < strlen(s); i++)
// determine letters by checking if its an alphabet and increase letter count by 1 for every letter detected
    {
        if (isalpha(s[i]))
        {
            number_letters++;
        }
// determine words by checking for blank spaces. characters after blank spaces indicate the begining of a new word, hence word count will be increased by 1

        if ((i == 0 && s[i] != ' ') || (i != strlen(s) - 1 && s[i] == ' ' && s[i + 1] != ' '))

        {
            number_words++;
        }

// determine what are sentences by using ". ! and ?" as indicators of end of sentences. When a sentence is detected increment the sentence count by 1
        if (s[i] == '.' || s[i] == '!' || s[i] == '?')

        {
            number_sentences++;
        }

    }

// solve for L, S and index respectively and ensure to cast the denominators and round the result of index to a whole number
    float L = (number_letters / (float)number_words) * 100;
    float S = (number_sentences / (float)number_words) * 100;
    int index = round(0.0588 * L - 0.296 * S - 15.8);


// if the result of index is less than 1 print Before grade 1, if greater than 16 print Grade 16+ anything else, the integer value of index should be printed alongside "grade"
    if (index <= 1)

    {
        printf("Before Grade 1\n");
    }

    else if (index >= 16)

    {
        printf("Grade 16+\n");
    }

    else

    {
        printf("Grade %i\n", index);
    }

}